apis/definition.yaml

```diff
@@ -38,11 +38,49 @@ spec:
                       description: Crossplane ProviderConfig to use for provisioning this resources
                       type: string
                       default: default
+                    vpcCidrBlock:
+                      type: string
+                      description: CIDR block for the VPC
+                      default: 192.168.0.0/16
+                    subnets:
+                      type: array
+                      items:
+                        type: object
+                        properties:
+                          availabilityZone:
+                            type: string
+                          type:
+                            type: string
+                            enum:
+                              - public
+                              - private
+                          cidrBlock:
+                            type: string
+                      description: Subnets to create in the VPC
+                      required:
+                        - availabilityZone
+                        - type
+                        - cidrBlock
+                      default:
+                        - availabilityZone: us-west-2a
+                          type: public
+                          cidrBlock: 192.168.0.0/18
+                        - availabilityZone: us-west-2b
+                          type: public
+                          cidrBlock: 192.168.64.0/18
+                        - availabilityZone: us-west-2a
+                          type: private
+                          cidrBlock: 192.168.128.0/18
+                        - availabilityZone: us-west-2b
+                          type: private
+                          cidrBlock: 192.168.192.0/18
                   required:
                     - deletionPolicy
                     - providerConfigName
                     - id
                     - region
+                    - subnets
+                    - vpcCidrBlock
```

Comment:
This will break backward compatibility and invalidate the usefulness of `default:` values. Is there any reason you decided to explicitly require this section?

Reply:
but a default is in definition - the API Server will Inject this default

Reply:
that's right, all good then!